app/src/main/java/io/github/gmathi/novellibrary/network/NovelApi_Novel.kt

```diff
@@ -6,15 +6,15 @@ import java.net.URI
 import java.util.regex.Pattern
 
 
-fun NovelApi.getNovelDetails(url: String): Novel? {
+fun NovelApi.getNovelDetails(url: String): Novel {
     var novel: Novel? = null
     val host = URI(url).host
     when {
         host.contains(HostNames.NOVEL_UPDATES) -> novel = getNUNovelDetails(url)
         host.contains(HostNames.ROYAL_ROAD) -> novel = getRRNovelDetails(url)
         host.contains(HostNames.WLN_UPDATES) -> novel = getWlnNovelDetails(url)
     }
-    return novel
+    return novel!!
```

Comment:
We don't want a NPE When it fails to get novel details. Always handle it safely.